### PR TITLE
[Docs] Adds list of available params and settings

### PR DESCRIPTION
**Pull request by bravomartin**
This PR updates the /docs to include the currently available params, their props and global settings.
We should eventually find a way to keep these in sync with the public website, but for now this makes them easily accessible to manually update the docs. 

From now on we should make sure `/docs` is whenever a change is made to the API.

### CODE REVIEW

File: doc/doc.md [hunk 1]
```diff
@@ -166,8 +167,82 @@ export const settings = {
...
+| options      | `['value']` or `[{label,value}]` or `{label: value, ...}` | undefined | If present, displays a dropdown with the provided options.                                                        |
+| validation   | `function`                                                | undefined | If present, executes function with the new value. Should return a string describing the error or null if no error |
+
+#### Image
+
+| Prop       | Type       | Default   | Description                                                                                                                   |
+| ---------- | ---------- | --------- | ----------------------------------------------------------------------------------------------------------------------------- |
+| type (\*)  | `"image"`  |           | Defines a param of type 'image'                                                                                               |
+| default    | `string`   | undefined | If present, defaults to this value.                                                                                           |
```
Comment (fdoflorenzano) [inline doc/doc.md:231]: I wouldn't list it here. There is no value you could give it for it to work right now.

File: doc/doc.md [hunk 2]
```diff
@@ -166,8 +167,82 @@ export const settings = {
...
+| options      | `['value']` or `[{label,value}]` or `{label: value, ...}` | undefined | If present, displays a dropdown with the provided options.                                                        |
+| validation   | `function`                                                | undefined | If present, executes function with the new value. Should return a string describing the error or null if no error |
+
+#### Boolean or Toggle
+
+| Prop         | Type        | Default   | Description                                                                                                       |
+| ------------ | ----------- | --------- | ----------------------------------------------------------------------------------------------------------------- |
+| type (\*)    | `"boolean"` |           | Defines a param of type 'boolean'                                                                                 |
+| default (\*) | `string`    |           | Default to this value.                                                                                            |
```
Comment (fdoflorenzano) [inline doc/doc.md:211]: Type should be boolean right?

File: doc/doc.md [hunk 3]
```diff
@@ -166,8 +167,82 @@ export const settings = {
...
+
+---
+
+### Available Settings
+
+These are the available settings:
+
+| Setting     | Value                       | Default   | Description                                                                                                                                                                                                       |
+| ----------- | --------------------------- | --------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| engine (\*) | `svg\|canvas\|react\|d3`    | undefined | Determines the engine to handle the design function                                                                                                                                                               |
```
Comment (fdoflorenzano) [inline doc/doc.md:244]: I feel this gives the impression that it should be a string, and one of those values, but it's a but more complex than that.

**Review by fdoflorenzano — CHANGES_REQUESTED** [review]
Commented on some weird stuff.

File: doc/doc.md [hunk 4]
```diff
@@ -166,8 +167,82 @@ export const settings = {
...
+| `color`   | Color picker           | Color `string`       |
+| `image`   | File selector          | `File` object        |
+
+#### Text
+
+| Prop         | Type                                              | Default   | Description                                                                                                       |
+| ------------ | ------------------------------------------------- | --------- | ----------------------------------------------------------------------------------------------------------------- |
+| type (\*)    | `"text"`                                          |           | Defines a param of type 'text'                                                                                    |
+| default (\*) | `string`                                          |           | Default to this value.                                                                                            |
+| editable     | `boolean`                                         | true      | If false, disables the field in the UI.                                                                           |
```
[the remark this answers is not in the document]
Comment (bravomartin) [reply doc/doc.md:188]: 👍

File: doc/doc.md [hunk 5]
```diff
@@ -166,8 +167,82 @@ export const settings = {
...
+| `number`  | Number field or slider | `number`             |
+| `boolean` | Toggle                 | `boolean`            |
+| `color`   | Color picker           | Color `string`       |
+| `image`   | File selector          | `File` object        |
+
+#### Text
+
+| Prop         | Type                                              | Default   | Description                                                                                                       |
+| ------------ | ------------------------------------------------- | --------- | ----------------------------------------------------------------------------------------------------------------- |
+| type (\*)    | `"text"`                                          |           | Defines a param of type 'text'                                                                                    |
```
[the remark this answers is not in the document]
Comment (bravomartin) [reply doc/doc.md:186]: 👍

File: doc/doc.md [hunk 6]
```diff
@@ -166,8 +167,82 @@ export const settings = {
...
+| default (\*) | `string`                                          |           | Default to this value.                                                                                            |
+| editable     | `boolean`                                         | true      | If false, disables the field in the UI.                                                                           |
+| options      | `['value']\|[{label,value}]\|{label: value, ...}` | undefined | If present, displays a dropdown with the provided options.                                                        |
```
[the remark this answers is not in the document]
Comment (bravomartin) [reply doc/doc.md:189]: 👍

File: doc/doc.md (hunk 1 above)
Comment (bravomartin) [reply doc/doc.md:231]: 👍

File: doc/doc.md (hunk 2 above)
Comment (bravomartin) [reply doc/doc.md:211]: also while typing this i thought this should be optional ?. like no default == false ?

File: doc/doc.md (hunk 3 above)
Comment (bravomartin) [reply doc/doc.md:244]: how would you explain it?

**Review by fdoflorenzano — APPROVED** [review]

**Pull request merged**